### PR TITLE
GraphiQL component: `forcedTheme `

### PR DESCRIPTION
**Pull request by TuvalSimha**
closes issue: #3398 

Description:

- User can force theme - added `forcedTheme` prop
- When `forcedTheme` is set, the toggle theme option in the setting disappears.

### CODE REVIEW

**Review by B2o5T — CHANGES_REQUESTED** [review]
> Can enforce the theme from editorTheme prop.

so if the user can enforce the theme via `editorTheme` why do we need additional prop `showThemeSettings`?

We could show the theme toggle only if `props.editorTheme` was not set

Also please add simple cypress tests

**Comment by TuvalSimha** [comment]
But if you want to implement you theme, but still give users the option
change the theme, how you can do it?

**Comment by mvdiener** [comment]
Hello, I'm not an active contributor to this repo, but I am actively watching these changes because I'm very interested in utilizing this property.

I'm wondering if these changes still allow for the utilization of the system theme? In the code for the `useTheme` [hook](https://github.com/graphql/graphiql/blob/main/packages/graphiql-react/src/theme.ts#L10), the theme value of null indicates the theme should use the system default. Does the new `forcedTheme` property allow for the scenario where we want to hide the theme toggle in the settings modal, but also want the theme to be set based on the system default?

Perhaps a third option for theme: `forcedTheme?: 'light' | 'dark' | 'system';` and under the hood, 'system' can be translated to null when calling `setTheme()`?

Thank you, and looking forward to these upcoming changes!

**Comment by B2o5T** [comment]
@mvdiener i guess we can use `undefined` value to disable `forcedTheme` and `null` to set as theme `system` and migrate to next-themes package in v4 which has `system` that is aka `null` now (in our useTheme hook)

File: packages/graphiql/cypress/e2e/theme.cy.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+describe('Theme', () => {
+  it('Switches to light theme when forceTheme is light', () => {
+    cy.visit('/?query={test}&forceTheme=light');
```
Comment (B2o5T) [inline packages/graphiql/cypress/e2e/theme.cy.ts:3]:
```suggestion
    cy.visit('/?query={test}&forcedTheme=light');
```

File: packages/graphiql/cypress/e2e/theme.cy.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+describe('Theme', () => {
+  it('Switches to light theme when forceTheme is light', () => {
```
Comment (B2o5T) [inline packages/graphiql/cypress/e2e/theme.cy.ts:2]:
```suggestion
  it('Switches to light theme when forcedTheme is light', () => {
```

File: packages/graphiql/cypress/e2e/theme.cy.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    cy.visit('/?query={test}&forceTheme=light');
+    cy.get('body').should('have.class', 'graphiql-light'); // Check for the class indicating light theme
+  });
+
+  it('Switches to dark theme when forceTheme is dark', () => {
+    cy.visit('/?query={test}&forceTheme=dark');
```
Comment (B2o5T) [inline packages/graphiql/cypress/e2e/theme.cy.ts:8]:
```suggestion
  it('Switches to dark theme when forcedTheme is dark', () => {
    cy.visit('/?query={test}&forcedTheme=dark');
```

File: packages/graphiql/cypress/e2e/theme.cy.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    cy.get('body').should('have.class', 'graphiql-dark'); // Check for the class indicating dark theme
+  });
+
+  it('Defaults to light theme when forceTheme value is invalid', () => {
+    cy.visit('/?query={test}&forceTheme=invalid');
```
Comment (B2o5T) [inline packages/graphiql/cypress/e2e/theme.cy.ts:13]:
```suggestion
  it('Defaults to light theme when forcedTheme value is invalid', () => {
    cy.visit('/?query={test}&forcedTheme=invalid');
```

File: packages/graphiql/resources/renderExample.js [hunk 5]
```diff
@@ -102,5 +102,6 @@ root.render(
     shouldPersistHeaders: true,
     inputValueDeprecation: GraphQLVersion.includes('15.5') ? undefined : true,
     onTabChange,
+    forceTheme: parameters.forceTheme,
```
Comment (B2o5T) [inline packages/graphiql/resources/renderExample.js:105]:
```suggestion
    forcedTheme: parameters.forcedTheme,
```

File: packages/graphiql/src/components/GraphiQL.tsx [hunk 6]
```diff
@@ -209,6 +211,12 @@ export type GraphiQLInterfaceProps = WriteableEditorProps &
      * settings modal.
      */
     showPersistHeadersSettings?: boolean;
+    /**
+     * forcedTheme allows to enforce a specific theme for GraphiQL.
+     * This is useful when you want to make sure that GraphiQL is always
+     * rendered with a specific theme, even if the user has a different
```
Comment (B2o5T) [inline packages/graphiql/src/components/GraphiQL.tsx:217]:
```suggestion
     * rendered with a specific theme
```

File: packages/graphiql/src/components/GraphiQL.tsx [hunk 7]
```diff
@@ -760,39 +782,41 @@ export function GraphiQLInterface(props: GraphiQLInterfaceProps) {
               </ButtonGroup>
             </div>
           ) : null}
-          <div className="graphiql-dialog-section">
-            <div>
-              <div className="graphiql-dialog-section-title">Theme</div>
-              <div className="graphiql-dialog-section-caption">
-                Adjust how the interface looks like.
+          {props.forcedTheme ? null : (
```
Comment (B2o5T) [inline packages/graphiql/src/components/GraphiQL.tsx:785]:
```suggestion
          {!props.forcedTheme && (
```

File: .changeset/famous-shirts-mate.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+'graphiql': minor
+---
+
+Create new prop to GraphiQL component: `forcedTheme` to control the theme.
```
Comment (B2o5T) [inline .changeset/famous-shirts-mate.md:5]:
```suggestion
Add a new prop to GraphiQL component: `forcedTheme` to force the theme and hide the theme switcher.
```

File: packages/graphiql/cypress/e2e/theme.cy.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
+describe('Theme', () => {
+  it('Switches to light theme when forcedTheme is light', () => {
+    cy.visit('/?query={test}&forcedTheme=light');
+    cy.get('body').should('have.class', 'graphiql-light'); // Check for the class indicating light theme
```
Comment (B2o5T) [inline packages/graphiql/cypress/e2e/theme.cy.ts:4]:
```suggestion
    cy.get('body').should('have.class', 'graphiql-light');
```

File: packages/graphiql/cypress/e2e/theme.cy.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+  });
+
+  it('Switches to dark theme when forcedTheme is dark', () => {
+    cy.visit('/?query={test}&forcedTheme=dark');
+    cy.get('body').should('have.class', 'graphiql-dark'); // Check for the class indicating dark theme
```
Comment (B2o5T) [inline packages/graphiql/cypress/e2e/theme.cy.ts:9]:
```suggestion
    cy.get('body').should('have.class', 'graphiql-dark');
```

File: packages/graphiql/cypress/e2e/theme.cy.ts [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+  });
+
+  it('Defaults to light theme when forcedTheme value is invalid', () => {
+    cy.visit('/?query={test}&forcedTheme=invalid');
+    cy.get('body').should('have.class', 'graphiql-light'); // Check for the class indicating light theme
```
Comment (B2o5T) [inline packages/graphiql/cypress/e2e/theme.cy.ts:14]:
```suggestion
    cy.get('body').should('have.class', 'graphiql-light');
```

**Comment by louisscruz** [comment]
Are there any additional adjustments needed to get this in? cc @B2o5T

File: packages/graphiql/src/components/GraphiQL.tsx [hunk 12]
```diff
@@ -209,6 +211,12 @@ export type GraphiQLInterfaceProps = WriteableEditorProps &
      * settings modal.
      */
     showPersistHeadersSettings?: boolean;
+    /**
+     * forcedTheme allows to enforce a specific theme for GraphiQL.
```
Comment (TallTed) [inline packages/graphiql/src/components/GraphiQL.tsx:215]:
```suggestion
     * forcedTheme allows enforcement of a specific theme for GraphiQL.
```

File: packages/graphiql/src/components/GraphiQL.tsx [hunk 13]
```diff
@@ -760,39 +782,41 @@ export function GraphiQLInterface(props: GraphiQLInterfaceProps) {
...
-            <div>
-              <div className="graphiql-dialog-section-title">Theme</div>
-              <div className="graphiql-dialog-section-caption">
-                Adjust how the interface looks like.
+          {!props.forcedTheme && (
+            <div className="graphiql-dialog-section">
+              <div>
+                <div className="graphiql-dialog-section-title">Theme</div>
+                <div className="graphiql-dialog-section-caption">
+                  Adjust how the interface looks like.
```
Comment (TallTed) [inline packages/graphiql/src/components/GraphiQL.tsx:790]:
```suggestion
                  Adjust how the interface appears.
```

**Review by dimaMachina — APPROVED** [review]

**Pull request merged**